### PR TITLE
Be more specific when considering shards.

### PR DESCRIPTION
**Pull request by timuralp**
When we process container shards, we need to be more exact about the
matching containers. This removes the extraneous containers from being
considered (for example, previously, a shard of <container>_segments
would be processed as a shard of <container>).

### CODE REVIEW

File: container_crawler/crawler.py [hunk 1]
```diff
@@ -471,17 +473,19 @@ def _enqueue_sharded_container(self, settings, per_account=False,
         """
         # TODO: look into saving the sharded state of the container
         sharded_account = '.shards_' + settings['account']
-        sharded_container = settings['container']
+        prefix = '{}-'.format(settings['container'])
         all_sharded_containers = self.list_containers(
-            sharded_account, prefix=sharded_container)
+            sharded_account, prefix=prefix)
+        shard_re = re.compile(SHARD_REGEX.format(settings['container']))
```
Comment (tipabu) [inline container_crawler/crawler.py:479]: What if the container name has characters that could be part of a regex? Should this be

    shard_re = re.compile(SHARD_REGEX.format(re.escape(settings['container'])))

?

File: container_crawler/crawler.py [hunk 2]
```diff
@@ -471,17 +473,19 @@ def _enqueue_sharded_container(self, settings, per_account=False,
...
+            if not shard_re.match(container):
+                continue
             settings_copy = settings.copy()
             settings_copy['internal_account'] = sharded_account
             settings_copy['internal_container'] = container
             self._enqueue_container(settings_copy, per_account, job)
         self._prune_deleted_containers(
-            sharded_account, all_sharded_containers,
-            prefix=sharded_container)
+            sharded_account, all_sharded_containers, prefix=prefix)
```
Comment (tipabu) [inline container_crawler/crawler.py:488]: This part didn't need updating, too?

File: container_crawler/crawler.py (hunk 1 above)
Comment (timuralp) [reply container_crawler/crawler.py:479]: Good point -- will add a test

File: container_crawler/crawler.py (hunk 2 above)
Comment (timuralp) [reply container_crawler/crawler.py:488]: The `_prune_status_files()` change is here: https://github.com/swiftstack/container-crawler/pull/49/commits/222ba736e1272253da087174eb0ea71028c0c576

**Pull request merged**